From 94497eaf2e8f2bc0c61d2aeaf40e0a21bfef83ef Mon Sep 17 00:00:00 2001
From: Maher Khenissi <maherx.khenissi@intel.com>
Date: Thu, 3 Jan 2013 10:24:16 +0100
Subject: [PORT FROM MAIN] Tethering/wifi: fix klocwork issues

BZ: 77624

1) Pointer 'mAddrs' returned from call to function 'malloc'
at line 129 may be NULL and may be dereferenced at line 132.
2) Pointer 'mDnsForwarders' checked for NULL at line 341
may be dereferenced at line 369.

Feature   : Wlan
Category  : Improvement
Purpose   : AOSP+
Origin    : Upstream
Upstream-candidate: no

Change-Id: I1df3f1a3b32b016031b8c7ff92dcc729a6cebcc8
Orig-Change-Id: I2fbc946ead4feb423ca980938739a205f039e9e6
Signed-off-by: Maher Khenissi <maherx.khenissi@intel.com>
---
 TetherController.cpp |    9 ++++++++-
 1 files changed, 8 insertions(+), 1 deletions(-)

diff --git a/TetherController.cpp b/TetherController.cpp
index 0214326..a3bb1b7 100644
--- a/TetherController.cpp
+++ b/TetherController.cpp
@@ -126,6 +126,12 @@ int TetherController::startTethering(int num_addrs, struct in_addr* addrs) {
 
         /* Copy the new addrs*/
         mAddrs = (struct in_addr*)malloc(sizeof (struct in_addr*) * num_addrs);
+        if (!mAddrs) {
+            ALOGE("malloc failed (%s)", strerror(errno));
+            close(pipefd[0]);
+            close(pipefd[1]);
+            return -1;
+        }
         mNum_addrs = num_addrs;
         for (int addrIndex=0; addrIndex < mNum_addrs;) {
             mAddrs[addrIndex] = addrs[addrIndex];
@@ -296,7 +302,8 @@ int TetherController::resetDnsForwarders() {
     LOGD("(resetDnsForwarders) Sending update msg to dnsmasq [%s]", daemonCmd);
     if (write(mDaemonFd, daemonCmd, strlen(daemonCmd) +1) < 0) {
         LOGE("(resetDnsForwarders) Failed to send update command to dnsmasq (%s)", strerror(errno));
-        mDnsForwarders->clear();
+        if (mDnsForwarders != NULL)
+            mDnsForwarders->clear();
         return -1;
     }
 
-- 
1.7.4.1

